### DISCUSSION TITLE
How to do inferencing with Yolov5 in TPUv3.8?

### DISCUSSION THREAD

**Discussion by kurianbenoy-sentient** [ultralytics/yolov5#7696] [Q&A]
Hi,

When I was looking through the github repository, I couldn't find the weight's for `yolov5s_edgetpu.tflite`. Can someone point me where it's available?

With that `yolov5s_edgetpu.tflite` model, is it possible to do inference using TPUv3.8? Or is there any recommended way to do inferencing with Yolov5 in Pytorch itself. Using softwares like `torch_xla` it looks like a lot of code change is required to train like that. Is my assumption correct?

**Comment by glenn-jocher** [comment]
@kurianbenoy-sentient see **TFLite, ONNX, CoreML, TensorRT Export** tutorial for generating Edge TPU YOLOv5 models:

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️  RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Roboflow for Datasets, Labeling, and Active Learning](https://github.com/ultralytics/yolov5/issues/4975)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [Architecture Summary](https://github.com/ultralytics/yolov5/issues/6998)&nbsp; ⭐ NEW

Good luck 🍀 and let us know if you have any other questions!

**Reply by kurianbenoy-sentient** [reply]
@glenn-jocher do you have any recommendations on inferencing with Yolov5 model in TPUs?

**Reply by glenn-jocher** [reply]
@kurianbenoy-sentient follow the export tutorial

**Reply by kurianbenoy-sentient** [reply]
I did follow the export models, yet nothing is mentioned on running TPU models. Should I use model `TensorFlow Edge TPU` for exporting models to run inference in TPUs? @glenn-jocher

**Reply by glenn-jocher** [reply]
@kurianbenoy-sentient this might help: https://github.com/ultralytics/yolov5/discussions/3428

**Reply by kurianbenoy-sentient** [reply]
According to that competition winner EdgeTPUs and normal TPUs are totally different. So I don't think the link @glenn-jocher shared works.

I asked same question to winner of that competition few days back:

> . The EdgeTPU and TPU are very different platforms. There is already a Keras implementation of the model so that would be a good start to getting it running.

https://github.com/jveitchmichaelis/edgetpu-yolo/issues/5
